### PR TITLE
Add help text for session's client address

### PR DESCRIPTION
**Pull request by kroepke**
The auth framework does not support updating sessions beyond the last used timestamp.
Especially the host is not resettable after creation.

Users might expect the address to change, but if we want to expose that we need to save this information separately in the future.

This change merely informs the user about this circumstance and makes the headers available internally for some optimization regarding session extension (less DB requests).

Refs #2656

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/shared/security/ShiroSecurityContextFilter.java [hunk 1]
```diff
@@ -44,10 +45,13 @@
 
 @Priority(Priorities.AUTHENTICATION)
 public class ShiroSecurityContextFilter implements ContainerRequestFilter {
+    public static final String REQUEST_HEADERS = "REQUEST_HEADERS";
     private final DefaultSecurityManager securityManager;
     private Provider<Request> grizzlyRequestProvider;
     private final Set<IpSubnet> trustedProxies;
 
+    public static final String X_GRAYLOG_REMOTE_ADDRESS = "x-graylog-remote-address";
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/shared/security/ShiroSecurityContextFilter.java]: What is this constant being used for?

File: graylog2-web-interface/src/components/users/UserList.jsx [hunk 2]
```diff
@@ -90,11 +105,13 @@ const UserList = React.createClass({
       </OverlayTrigger>);
     }
 
-    const roleBadges = user.roles.map((role) => <span key={role} className={`${UserListStyle.roleBadgeFixes} label label-${role === 'Admin' ? 'info' : 'default'}`} >{role}</span>);
+    const roleBadges = user.roles.map((role) => <span key={role}
+                                                      className={`${UserListStyle.roleBadgeFixes} label label-${role === 'Admin' ? 'info' : 'default'}`}>{role}</span>);
 
     let actions = null;
     if (user.read_only) {
-      const tooltip = <Tooltip id="system-user">System users can only be modified in the Graylog configuration file.</Tooltip>;
```
Comment (joschi) [inline graylog2-web-interface/src/components/users/UserList.jsx]: Unnecessary whitespace change.

File: graylog2-web-interface/src/components/users/UserList.jsx [hunk 3]
```diff
@@ -90,11 +105,13 @@ const UserList = React.createClass({
       </OverlayTrigger>);
     }
 
-    const roleBadges = user.roles.map((role) => <span key={role} className={`${UserListStyle.roleBadgeFixes} label label-${role === 'Admin' ? 'info' : 'default'}`} >{role}</span>);
```
Comment (joschi) [inline graylog2-web-interface/src/components/users/UserList.jsx]: Unnecessary whitespace change.

File: graylog2-server/src/main/java/org/graylog2/shared/security/ShiroSecurityContextFilter.java (hunk 1 above)
Comment (kroepke) [inline graylog2-server/src/main/java/org/graylog2/shared/security/ShiroSecurityContextFilter.java]: Apparently left over. If unused I'll remove it.

**Comment by joschi** [comment]
LGTM. 👍
